pdns/dnsdist-lua-bindings-dnsquestion.cc

```diff
@@ -63,6 +63,27 @@ void setupLuaBindingsDNSQuestion()
 
       return *dq.ednsOptions;
     });
+  g_lua.registerFunction<std::string(DNSQuestion::*)(void)>("getTrailingData", [](const DNSQuestion& dq) {
+      const char* message = reinterpret_cast<const char*>(dq.dh);
+      const uint16_t messageLen = getDNSPacketLength(message, dq.len);
+      const std::string tail = std::string(message + messageLen, dq.len - messageLen);
+      return tail;
+    });
+  g_lua.registerFunction<bool(DNSQuestion::*)(std::string)>("setTrailingData", [](DNSQuestion& dq, const std::string& tail) {
+      char* message = reinterpret_cast<char*>(dq.dh);
+      const uint16_t messageLen = getDNSPacketLength(message, dq.len);
+      const uint16_t tailLen = tail.size();
+      if(messageLen + tailLen > dq.size) {
```

Comment:
Not a big deal but it looks like `messageLen` + `tailLen` could overflow before being promoted. Perhaps something like:
```suggestion
      if(tailLen > (dq.size - messageLen)) {
```

Reply:
Nice catch! Updated.